src/test/java/use_cases/turn_start_use_case/TurnStartInteractorTest.java

```diff
@@ -0,0 +1,257 @@
+
+        Player player1 = playerFactory.createPlayer("Player 1", new ArrayList<>(List.of(creature1)),
+                (PlayerDeck) playerDeck1, (EssenceDeck) essenceDeck1);
+        Player player2 = playerFactory.createPlayer("Player 2", new ArrayList<>(List.of(creature2)),
+                (PlayerDeck) playerDeck2, (EssenceDeck) essenceDeck2);
+
+        // Leave Player 1 with an empty hand, but give Player 2 a hand of 9 cards.
+        for (int i = 0; i < 9; i++) {
+            player2.addCard(player2.getEssenceDeck().draw());
+        }
+        gameState = new GameState();
+        gameState.setPlayers(player1, player2);
+    }
+
+    @Test
+    void testDrawCards_BelowHandCapacity() {
+        TurnStartOutputBoundary presenter = new TurnStartOutputBoundary() {
+            @Override
+            public DrawCardOutputModel showDrawResult(DrawCardOutputModel outputData) {
+                assertEquals(4, outputData.getNewCardIds().size(),
+                        "2 play deck cards and 2 Essence cards should be drawn.");
+                assertEquals(0, outputData.getBurntCardIds().size(),
+                        "No cards should be discarded because hand is not full.");
+                return null;
+            }
+
+            @Override
+            public CreatureStatsUpdateModel showClearBuffs(CreatureStatsUpdateModel outputData) {
+                fail("drawCards() should not call showClearBuffs().");
+                return null;
+            }
+
+            @Override
+            public TriggerEffectUpdateModel showEffectUpdates(TriggerEffectUpdateModel outputData) {
+                fail("drawCards() should not call showEffectUpdates().");
+                return null;
+            }
+        };
+        TurnStartInteractor interactor = new TurnStartInteractor(gameState, presenter);
+        interactor.drawCards();
+    }
+
+    @Test
+    void testDrawCards_ExceedHandCapacity() {
+        TurnStartOutputBoundary presenter = new TurnStartOutputBoundary() {
+            @Override
+            public DrawCardOutputModel showDrawResult(DrawCardOutputModel outputData) {
+                assertEquals(1, outputData.getNewCardIds().size(),
+                        "1 play deck card should be drawn.");
+                assertEquals(1, outputData.getNewCardIds().size(),
+                        "The card drawn should have id 1, which comes from the PlayerDeck, not EssenceDeck.");
+                assertTrue(outputData.getNewCardNames().contains("name1"),
+                        "The output should contain the name of the card.");
+                assertEquals(3, outputData.getBurntCardIds().size(),
+                        "3 cards should be discarded because hand is full.");
+                assertTrue(outputData.getBurntCardNames().contains("Essence"),
+                        "There should be cards with name 'Essence' in the discarded cards.");
+                return null;
```

Comment:
Since we discard 3 cards, two of them is essence. I believe that it would be good to add another test just to check to make sure a card from the PlayerDeck was discarded.

Reply:
yeah! that would go like this
```suggestion
                assertTrue(outputData.getBurntCardNames().contains("Essence"),
                        "There should be cards with name 'Essence' in the discarded cards.");
                assertTrue(outputData.getBurntCardNames().contains("name1"),
                        "There should be a card with name 'name1' in the discarded cards.");
                return null;
```